docs/node-operators/setup/generate-keys.md

```diff
@@ -1,58 +1,139 @@
+title: Generate private keys
 ---
 
 import extension from './assets/extension.png';
 import mobile from './assets/mobile.png';
 
-# Generate keys
+Each validator node requires private keys for signing blocks and consensus messages sent to the other validators in the network.
 
-This guide demonstrates how to generate private keys for your nodes.
+The required private keys include one ECDSA key and one BLS key. The keys are named after digital signature schemes: Elliptic Curve Digital Signature Algorithm (ECDSA) and Boneh–Lynn–Shacham (BLS), respectively.
 
-## Generate locally
+- The ECDSA key is used for signing blocks.
+- The BLS key is used for producing finality votes. Validators who produce finality vote receive a reward. If a validator fails to produce a finality vote, no slashing penalty applies.
 
-### Prerequisites
+## Generate BLS keys
+
+You can generate an ECDSA key pair using the Ronin CLI or your node's Docker image.
+
+### Generate using the CLI
+
+1. Build Ronin from the source code:
+
+    ```
+    > git clone https://github.com/axieinfinity/ronin
+    > cd ronin
+    > make ronin
+    ```
+
+2. Add the Ronin binary path to the `$PATH` environment variable:
+
+    ```
+    > cp ./build/bin/ronin /usr/local/bin
+    ```
+
+3. Generate a BLS key pair:
+
+    ```
+    > mkdir bls_keystore
+    > echo "input_your_password" > bls_password
+    > ronin account generatebls --secret
+
+    Successfully generated BLS key
+    Public key: {your_public_key}
+    Secret key: {your_private_key}
+    ```
+
+    An `all-accounts.keystore.json` file containing the encrypted BLS key is created inside the `bls_keystore` directory.
+
+    To reveal the public key, run this command:
+
+    ```
+    > ronin account listbls
+
+    BLS public key #0: {your_public_key}
+    ```
+
+    To reveal both the public key and private key, run this command:
+
+    ```
+    > ronin account listbls --secret
+
+    BLS public key #0: {your_public_key}
+    BLS secret key #0: {Secret key: {your_private_key}
+    }
+    ```
+
+### Generate using Docker image
+
+In the `.env` file of your node's Docker image, configure the following:
+
+1. Leave `BLS_PRIVATE_KEY` empty but enter the password in `BLS_PASSWORD`.
+2. Set `BLS_AUTO_GENERATE` to `true`, and then run `docker-compose up -d`.
+3. Run `docker-compose logs node`, which returns `Using BLS account your-public-key`.
+
+If you want to view the BLS private key, do the following:
+
+1. Set `BLS_SHOW_PRIVATE_KEY=true`, run `docker-compose up -d`
+2. Run `docker-compose logs node`, which returns `BLS secret key #0: {your_private_key}`. **Note:** This command just shows the private key without running the node.
+3. Set `BLS_SHOW_PRIVATE_KEY` to `false`, and then run `docker-compose up -d` to start the node.
+
+For a non-validator node, add the following variables to the `.env` file:
+
+```
+ENABLE_FAST_FINALITY=true
+ENABLE_FAST_FINALITY_SIGN=false
+```
```

Comment:
This seems unnecessary in the key generation section, non-validator does not need to generate key